extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -203,34 +204,59 @@ func CreateRKE2K3SSnapshot(client *rancher.Client, clusterName string) error {
 }
 
 // RestoreRKE1Snapshot is a helper function to restore a snapshot on an RKE1 cluster. Returns error if any.
-func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRestore *management.RestoreFromEtcdBackupInput) error {
+func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRestore *management.RestoreFromEtcdBackupInput, initialControlPlaneValue, initialWorkerValue string) error {
 	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
 	if err != nil {
 		return err
 	}
 
-	clusterResp, err := client.Management.Cluster.ByID(clusterID)
+	cluster, err := client.Management.Cluster.ByID(clusterID)
+	if err != nil {
+		return err
+	}
+
+	updatedCluster := cluster
+
+	updatedCluster.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableControlplane = initialControlPlaneValue
+	updatedCluster.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableWorker = initialWorkerValue
+
+	_, err = client.Management.Cluster.Update(cluster, updatedCluster)
 	if err != nil {
 		return err
 	}
 
 	logrus.Infof("Restoring snapshot: %v", snapshotRestore.EtcdBackupID)
-	err = client.Management.Cluster.ActionRestoreFromEtcdBackup(clusterResp, snapshotRestore)
+	err = client.Management.Cluster.ActionRestoreFromEtcdBackup(cluster, snapshotRestore)
 	if err != nil {
 		return err
 	}
 
+	err = wait.PollUntilContextTimeout(context.TODO(), 5*time.Second, defaults.FiveMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
Maybe I don't see it, but doesn't it need to check this poll error?

Reply:
good point! Just return the err instead of nil

Reply:
You don't want the code to feel risky? Wow...lol. Nice catch, I updated it.